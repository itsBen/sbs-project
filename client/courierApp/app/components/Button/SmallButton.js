import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

export default ({ title, onPress, buttonStyle }) => (
  <TouchableOpacity onPress={onPress}>
    <View
      style={
        buttonStyle === 'white'
          ? [styles.container, styles.style_white__container]
          : styles.container
      }
    >
      <Text
        style={
          buttonStyle === 'white'
            ? [styles.title, styles.style_white__title]
            : styles.title
        }
      >
        {title}
      </Text>
    </View>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: 'rgb(230,37,101)',
    borderColor: 'rgb(230,37,101)',
    borderWidth: 1,
    paddingVertical: 5,
    paddingHorizontal: 10,
    borderRadius: 5,
  },
  title: {
    color: '#FFF',
    fontSize: 17,
  },
  style_white__container: {
    backgroundColor: '#fff',
    borderColor: 'transparent',
  },
  style_white__title: {
    color: '#808080',
  },
});
