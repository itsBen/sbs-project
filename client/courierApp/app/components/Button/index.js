import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

export { default as SmallButton } from './SmallButton';

export default ({ title, onPress, buttonStyle, disabled }) => (
  <TouchableOpacity onPress={onPress} disabled={disabled}>
    <View
      style={[
        styles[buttonStyle]
          ? styles[buttonStyle].container
          : styles.default.container,
        disabled && styles.default.disabledButton,
      ]}
    >
      <Text
        style={
          styles[buttonStyle] ? styles[buttonStyle].title : styles.default.title
        }
      >
        {title}
      </Text>
    </View>
  </TouchableOpacity>
);

const defaultStyles = {
  container: {
    alignItems: 'center',
    backgroundColor: 'rgb(230,37,101)',
    borderColor: 'rgb(230,37,101)',
    borderWidth: 1,
    paddingVertical: 10,
    paddingHorizontal: 10,
    borderRadius: 5,
  },
  title: {
    color: '#FFF',
    fontSize: 18,
  },
  disabledButton: {
    backgroundColor: '#c1c1c1',
    borderColor: 'transparent',
  },
};

const styles = {
  default: StyleSheet.create({ ...defaultStyles }),
  white: StyleSheet.create({
    container: {
      ...defaultStyles.container,
      backgroundColor: 'white',
      borderColor: '#808080',
    },
    title: {
      ...defaultStyles.title,
      color: '#808080',
    },
  }),
};
