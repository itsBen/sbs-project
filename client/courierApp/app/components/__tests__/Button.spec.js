/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button, { SmallButton } from '../Button';

it('renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with different buttonstyle', () => {
  const tree = renderer.create(<Button buttonStyle="white" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders small button correctly', () => {
  const tree = renderer.create(<SmallButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
