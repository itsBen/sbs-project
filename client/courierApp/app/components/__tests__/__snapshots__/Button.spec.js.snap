// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(230,37,101)",
          "borderColor": "rgb(230,37,101)",
          "borderRadius": 5,
          "borderWidth": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly with different buttonstyle 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#808080",
          "borderRadius": 5,
          "borderWidth": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#808080",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;

exports[`renders small button correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(230,37,101)",
        "borderColor": "rgb(230,37,101)",
        "borderRadius": 5,
        "borderWidth": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 17,
        }
      }
    />
  </View>
</View>
`;
